Lecture3/Task1,2,3,4/com/playtika/task1/lecture3/Main.java

```diff
@@ -0,0 +1,125 @@
+
+        Scanner in = new Scanner(System.in);
+        System.out.println("Here is a created knight!Fill in  Maximum and Minimum cost's value to find the corresponding ammunition! ");
+        try {
+            return getCostRange(in);
+        } catch (UserWrongInputException e) {
+            throw e;
+        }
+    }
+
+    private static CostRange getCostRange(Scanner in) throws UserWrongInputException {
+        try {
+            int minCost = in.nextInt();
+            int maxCost = in.nextInt();
+            CostRange costRange = new CostRange(minCost, maxCost);
+            System.out.println();
+            in.close();
+            return costRange;
+        } catch (InputMismatchException e) {
+            throw new UserWrongInputException("Please, fill in number!");
+        }
+    }
+
+    private static String buildSortedByUserInputResult(CostRange costRange, Armor firstSuite) {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<ArmorItem> listSelectedByCost = firstSuite.getByCost(costRange.getMinCost(), costRange.getMaxCost());
+        if (firstSuite.getByMinCost(costRange.getMinCost()).size() == 0) {
+            throw new WrongMinPriceException("MinPrice is too high!");
+        }
+        if (firstSuite.getByMaxCost(costRange.getMaxCost()).size() == 0) {
+            throw new WrongMaxPriceException("MaxPrice is too low!");
+        }
+        stringBuilder.append("Ammunition which falling in the sample! \n");
+        for (ArmorItem armorItem : listSelectedByCost) {
+            stringBuilder.append(armorItem);
+            stringBuilder.append("\n");
+        }
+        return stringBuilder.toString();
+    }
+
+    private static String buildKnightInfo(Knight firstKnight, Armor firstSuite) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("There is our knight:" + firstKnight + "\n");
+        stringBuilder.append("All his ammunition costs:" + firstSuite.getCost() + "\n");
+        stringBuilder.append("All his ammunition weight:" + firstSuite.getWeight() + "\n");
+        return stringBuilder.toString();
+    }
+
+    private static String buildSortedBysCostArmor(Armor firstSuite) {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<ArmorItem> sorted = firstSuite.sortedByCost();
+        stringBuilder.append("His ammunition sorted by cost in incremental order: \n");
+        for (ArmorItem armorItem : sorted) {
+            stringBuilder.append(armorItem + "\n");
+        }
+        return stringBuilder.toString();
+    }
+
+    private static void writeToFile(String text, boolean append) {
+        try (OutputStream writer = new FileOutputStream("Knight.txt", append)) {
+
+            writer.write(text.getBytes());
+            writer.flush();
+        } catch (IOException ex) { // if some problems with Knight.txt
+
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    private static String readFromFile() {
```

Comment:
I assume according to the task we need to `serialize`/`deserialize` the initial data